src/Plaid/PlaidClient.cs

```diff
@@ -52,15 +75,26 @@ public class PlaidClient
 			};
 			_baseUrl = $"https://{subDomain}.plaid.com/";
 
-			_secret = secret;
-			_clientId = clientId;
-			_accessToken = accessToken;
+			_secret = string.IsNullOrWhiteSpace(secret) ? null : secret;
+			_clientId = string.IsNullOrWhiteSpace(clientId) ? null : clientId;
+			_accessToken = string.IsNullOrWhiteSpace(accessToken) ? null : accessToken;
 			_apiVersion = apiVersion switch
 			{
 				ApiVersion.v20190529 => "2019-05-29",
 				_ => throw new ArgumentOutOfRangeException(nameof(ApiVersion), "Invalid API version provided."),
 			};
 
+			// Somewhat superfluous. Could just set _options to 'null' but the object is low-memory so 
```

Comment:
Since we have the fields, we really don't need the `_options` field at all. Remove the field and assignment in both constructors.

Reply:
In theory we should maybe remove the _fields and read from the _options where they are accessed now, and RequestBase.SetCredentials updated to take a PlaidOptions (or a second method to support both ways). ... but wasn't sure if that was superfluous refactoring.

The PlaidOptions should actually be exposed protected for subclasses since you can pass in and overridden PlaidOptions implementation to the dependency injected constructor and re-cast it in a subclass to pass additional settings around.

Reply:
`RequestBase.SetCredentials()` was refactored to support certain APIs that refuse to accept `ClientId` and `Secret`. 

Honestly, now that you mention it, `PlaidClient` should probably be `sealed`. I don't think there's any reason anyone should subclass it. As long as I stay up to date w/ the API, then add'l APIs won't be necessary, and there is no other value to this class besides interacting with the API.

Reply:
I just realized the implementation method are private... I have overridden some classes like this in the past simply to add tracing/logging on the calls for debugging - it was nice not to have to write a full wrapper and just override methods I wanted to log/debug so I would find it potentially annoying if it were sealed for any debugging like that... but functionally, unless the private implementation methods are made protected, there's not really a production function reason to extend it I guess.

Reply:
I can understand that. That said, if you look in the `PostAsync()` method, we're doing the logging there for all APIs, so there shouldn't be any other logging necessary, I wouldn't think. I'd entertain an argument otherwise though!

Also, I'm doing symbol nuget, so anyone should be able to debug directly into the methods even as a nuget package. I've verified this myself, so new technology for the win!

Reply:
I guess the argument against sealing it is, if someone needed to, and we made the implementations protected, they could re-write how it works if the library went out of date... but they could also Fork it I guess =)